static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx

```diff
@@ -44,23 +45,38 @@ const GroupEventAttachmentsFilter = (props: Props) => {
 
   return (
     <FilterWrapper>
-      <ButtonBar merged active={activeButton}>
-        <Button barId="all" size="sm" to={{pathname, query: allAttachmentsQuery}}>
-          {t('All Attachments')}
-        </Button>
-        {isMobilePlatform(project.platform) && (
-          <Button
-            barId="screenshot"
-            size="sm"
-            to={{pathname, query: onlyScreenshotQuery}}
-          >
-            {t('Screenshots')}
-          </Button>
-        )}
-        <Button barId="onlyCrash" size="sm" to={{pathname, query: onlyCrashReportsQuery}}>
-          {t('Only Crash Reports')}
-        </Button>
-      </ButtonBar>
+      <SegmentedControl
+        aria-label={t('Algorithm')}
+        size="sm"
+        value={activeButton}
+        onChange={key => {
+          switch (key) {
+            case 'screenshot':
+              router.replace({pathname, query: onlyScreenshotQuery});
+              break;
+            case 'onlyCrash':
+              router.replace({pathname, query: onlyCrashReportsQuery});
+              break;
+            case 'all':
+            default:
+              router.replace({pathname, query: allAttachmentsQuery});
+          }
+        }}
+      >
+        {[
+          <SegmentedControl.Item key="all">{t('All Attachments')}</SegmentedControl.Item>,
+          ...(isMobilePlatform(project.platform)
+            ? [
+                <SegmentedControl.Item key="screenshot">
+                  {t('Screenshots')}
+                </SegmentedControl.Item>,
+              ]
+            : []),
+          <SegmentedControl.Item key="onlyCrash">
+            {t('Only Crash Reports')}
+          </SegmentedControl.Item>,
+        ]}
```

Comment:
this is contrived isn't it lol

```suggestion
          <SegmentedControl.Item key="all">{t('All Attachments')}</SegmentedControl.Item>
          {isMobilePlatform(project.platform) && (
	          <SegmentedControl.Item key="screenshot">
	            {t('Screenshots')}
	          </SegmentedControl.Item>
          }
          <SegmentedControl.Item key="onlyCrash">
            {t('Only Crash Reports')}
          </SegmentedControl.Item
```

Reply:
It is! Unfortunately `react-aria` doesn't allow us to put `false` as a child inside `SegmentedControl`.  If we do it'll work fine, but there will be a TS error. I considered modifying `SegmentedControl`'s prop types to accept booleans, but worried there might be an error in the future (if `react-aria` suddenly changes the way they parse collection children). If you think it's fine, though, I'm down for doing that!